### ISSUE TITLE
Examples of basic server/client

### ISSUE THREAD

**Issue by conotto** [emiago/sipgo#6]
Hello,
Are there any examples available of basic server with authentication or a client ?
Besides the 2 examples that are there.

Thanks!

**Comment by emiago** [comment]
Hi connotto. 

I do not have ready yet, but I will throw something once is done. 
This is normally simple digest Auth and some similar example done via http can be copy pasted, but
anyway I will try to publish this to you this week.

Normally server needs to create challenge and add header
Client needs to parse challenge and then send via **Authentication**

Building client on other hand library needs more things. It was more focused on servers. Client handle is mostly for proxy request.

**Comment by emiago** [comment]
Hi @conotto 

I pushed example. Pls checkout AuthExample branch. 
https://github.com/emiago/sipgo/tree/AuthExample/example/auth

This is just with REGISTER req chalenge

Usage:
Client: `go run ./client -u alice -p alice -dst 127.0.0.1:5060`
Server:` go run ./server -u "alice:alice,bob:bob"`

Let me know for some feedback. I think this things could be part of library. 
For digest I used this lib: github.com/icholy/digest
